modules/market/src/main/java/com/opengamma/strata/market/MarketDataValue.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market;
+
+import java.time.LocalDate;
+
+import com.opengamma.strata.basics.market.MarketDataId;
+import com.opengamma.strata.basics.market.MarketDataKey;
+
+/**
+ * A single item of high-level market data identified by a key and valuation date.
+ * <p>
+ * This represents a single piece of market data.
+ * The market data has typically been calibrated, such as a curve or surface.
+ *
+ * @see MarketDataKey
+ * @see MarketDataId
+ * @param <T>  the type of this market data
+ */
+public interface MarketDataValue<T extends MarketDataValue<T>> {
+
+  /**
+   * Gets the key used to obtain this market data.
+   * <p>
+   * A {@link MarketDataKey} is used to identify market data within a given context.
+   * A system may have multiple market data values with the same key, and {@link MarketDataId}
+   * is used to uniquely identify each.
+   * <p>
```

Comment:
It would be good to give a more concrete example of what a key represents vs an ID. Something like "there is a single USD discounting curve used in a calculation, but the system can contain multiple USD discounting curves used for different asset classes / couterparties / whatever".